src/Microsoft.SqlTools.ServiceLayer/Scripting/ScripterCore.cs

```diff
@@ -453,7 +431,51 @@ internal IEnumerable<Declaration> GetCompletionsForToken(ParseResult parseResult
             return Resolver.FindCompletions(
                 parseResult, parserLine, parserColumn, metadataDisplayInfoProvider);
         }
-        #endregion
 
+        /// <summary>
+        /// Wrapper method that calls Resolver.FindCompletions
+        /// </summary>
+        /// <param name="objectName"></param>
+        /// <param name="schemaName"></param>
+        /// <param name="objectType"></param>
+        /// <param name="tempFileName"></param>
+        /// <returns></returns>
+        internal ScriptingScriptOperation CreateScriptOperation(string objectName, string schemaName, string objectType, string tempFileName)
+        {
+            // object that has to be scripted
+            ScriptingObject obj = new ScriptingObject {
+                Name = objectName,
+                Schema = schemaName,
+                Type = objectType
+            };
+
+            // scripting options
+            ScriptOptions options = new ScriptOptions {
+			    ScriptCreateDrop = "ScriptCreate",
+			    TypeOfDataToScript = "SchemaOnly",
+			    ScriptStatistics = "ScriptStatsNone",
+			    TargetDatabaseEngineEdition = "SqlServerEnterpriseEdition",
+			    TargetDatabaseEngineType = "SingleInstance",
+			    ScriptCompatibilityOption = "Script140Compat",
+                IncludeIfNotExists = true
+		    };
+
+            List<ScriptingObject> objList = new List<ScriptingObject>();
+            List<string> inclList = new List<string>();
+            inclList.Add(objectType);
+            objList.Add(obj);
+
+            // create parameters for the scripting operation
+            ScriptingParams parameters = new ScriptingParams {
+                FilePath = tempFileName,
+                ConnectionString = ConnectionService.BuildConnectionString(this.connectionInfo.ConnectionDetails),
```

Comment:
So this builds a completely new connection to a Server and does scripting from that? This is going to be really slow for some scenarios, e.g. Azure.
- We should definitely look at optimizing this in the future. This is a great candidate for reuse of the ObjectExplorerService's Server object. We should be able to get the object from that. Special casing script as from the OE viewlet is one option, but I'd suggest we go further and figure out a way to share a connection for all these operations so the metadata is warmed up and ready.
- For now this is OK, but I see the need to make this right in the future so that perf is acceptable.